README.md

```diff
@@ -105,57 +117,129 @@ inv -l
+- `base.html` : contains some extra html structure exposing a `content` block for our page's content, and wraps it around the header and footer.
+- `header.html` and `footer.html` : standard header and footer block that will appear on each page
+- `home.html`: the home page template (duh)
+
+#### 🚲 Javascript
+
+In order to add some interactivity to the project, we are using [Vue 3](https://v3.vuejs.org/api/) and some good old VanillaJS.
+The JS assets are compiled in a single `index.js` file that includes everything for every page. If the bundle size starts to grow a little bit too much, you might need to think about splitting it into separate files for each page.
+
+#### 🖼️ Style
+
+We are using the [DSFR](https://github.com/GouvernementFR/dsfr) to build our front-end components.
+
+In addition we have a nice litle set of CSS Utilities to quickly build custom components, inspired by bootstrap, most of its documentation
+lives in the css located in `theme/less/` and is built using [Stylemark](https://github.com/mpetrovich/stylemark), you can read the live documentation in `udata_front/theme/stylemark/` after building it using `npm run build-stylemark`.
+
+You can access the UI-kit documentation by visiting the following route : `/_stylemark/index.html`
+
+Whenever a components needs some special styling, you can find their corresponding definitions inside `theme/less/specific/<component>`,
+it's best if we can avoid having too much specific styling, but sometimes you just really need it.
+
+### 🛠️ Build tools
+
+This project uses [Parcel 2](https://github.com/parcel-bundler/parcel) to build and transform our source files into nice bundles for browsers.
+Its config can be found in the `.parcelrc` file, as well as the `package.json` entries.
+
+Parcel does multiple custom things in this project :
+
+- Transform the `.js` files into modern Javascript for browsers
+- Transform the `less` files into modern CSS using `PostCSS`
+- Bundle the `svg` used in the JS files
+- Copy the static assets when they change (config is in the `package.json`'s `staticFiles` key)
+- Build the Stylemark (UI-kit) documentation on file change
+
+### 🏭 Javascript architecture
+
+#### 🏔️ Vue mouting
+
+We are using the full build of VueJS that includes the compiler in order to compile templates directly in the browser.
+
+There is a single VueJS app (in `index.js`) that contains every component and plugins.
+However, this app is mounted multiple times, on each DOM node containing a `vuejs` class.
+
+This allows us to mount the app only where its needed, because each subsequent mount is more DOM to compile and thus has an impact on performance. Moreover, mounting to the smallest possible HTML allows us to prevent accidental XSS vulnerability by forbidding users to compile their content with the Vue engine.
```

Comment:
```suggestion
This allows us to mount the app only where it's needed, because each subsequent mount is more DOM to compile and thus has an impact on performance. Moreover, mounting to the smallest possible HTML allows us to prevent accidental XSS vulnerability by forbidding users to compile their content with the Vue engine.
```